### DISCUSSION TITLE
Pulsar-client-shaded[pulsar-client]模块生成的源码包是空的

### DISCUSSION THREAD

**Discussion by 3286360470** [apache/pulsar#20175] [Q&A] [answered]
pulsar源码打包时，pulsar-client-shaded[pulsar-client]模块打出来的pulsar-client-2.11.0-sources.jar包里是空的，如下图所示：
https://global.discourse-cdn.com/business5/uploads/streamnative1/optimized/1X/fd253dafed92d2fbfa985ca5c7bf9c00faf17ef2_2_690x312.png

需要如何配置pom文件才能保证打出来的源码包不是空的，否则业务代码里依赖pulsar-client包时，不能正常查看源码。

**Comment by tisonkun** [comment, marked as answer]
This is a known issue https://github.com/apache/pulsar/issues/19918 and has been fixed https://github.com/apache/pulsar/pull/19956 on the master branch. IIRC 3.0 will include this fix.

**Reply by 3286360470** [reply]
@tisonkun thanks for your reply, your answer perfectly solved my problem!
